status/condition_set.go

```diff
@@ -237,3 +239,60 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// objectImpl is an embedded struct type that embeds the upstream type
+type objectImpl struct {
+	unstructured.Unstructured
+}
+
+func FromUnstructured(nodeClass *unstructured.Unstructured) Object {
+	return &objectImpl{Unstructured: *nodeClass}
+}
+
+func (o *objectImpl) GetConditions() []Condition {
+	var conditions []Condition
+	c, found, err := unstructured.NestedSlice(o.Object, "status", "conditions")
+	if err != nil || !found {
+		return conditions
+	}
+	for _, condition := range c {
+		cond := condition.(map[string]interface{})
```

Comment:
Is there a way to perform a one-liner or some typed conversion into a `metav1.Condition` rather than having to grab the key from each field?